code/controllers/Processes/aircurrents.dm

```diff
@@ -0,0 +1,496 @@
+//air currents is a system that keep track of tiles where more than a defined % air was added in a cycle
+//it then check neighboring tiles to find a path going away from the source with the most (or least depending on direction) air up to a defined number of turfs and equalise the air on the whole path at once. this will have for impact of spreading atmospherics much faster, heat can also be shared this way
+//current will only stop once the air stabilised on the whole current line for a defined number of cycles
+//it will also priorize holes open to space since it does not actually work on space, only 1 air current per hole can be active and as such without a multiplier would be dramatically slower than same air exchange between 2 connected room
+
+#define MINIMUM_AIR_CURRENTS_STARTPRESSURE 10 //any gas exchange with less pressure than this is just ignored if there is no currents in the area
+#define MINIMUM_AIR_CURRENTS_EXTRAPRESSURE 3 //any gas exchange with less pressure than this is just ignored if there are currents in the area
+#define MINIMUM_AIR_CURRENTS_STARTRATIO 0.3 //minimum ratio of pressure added to a tile in a cycle before its considered for the currents system when its the first
+#define MINIMUM_AIR_CURRENTS_EXTRARATIO 0.1 //minimum ratio of pressure needed for subsequent currents
+#define AIR_CURRENTS_LENGHT 8 //maximum number of turf considered for a current
+#define AIR_CURRENTS_STABLE_CYCLE 3 //number of cycles where no air was exchanged by the current before it can be considered dead
+#define AIR_CURRENTS_TOOSLOW_RATIO 0.09 //ratio at which its considered a slow currents and should be removed soon. should be slower than MINIMUM_AIR_CURRENTS_STARTRATIO
+#define AIR_CURRENTS_TOOSLOW_CYCLE 10 //number of cycles where less than AIR_CURRENTS_TOOSLOW_RATIO pressure was added to the air current tile before we are considered done
+#define AIR_CURRENTS_SPEED_FACTOR 1.0//1.0 cause the air in the current and the source to equilibrate 1 air_master tick in a closed room
+#define AIR_CURRENTS_TEMPERATURE_SPEED 0.0001 //affect how fast the temperature spread using the air currents
+#define AIR_CURRENTS_PRESSUREPUSH_MULT 2 //due to how the air is averaged on the chain, the pressure difference is actually often much smaller than it should be, so we simply apply add a multiplier
+#define AIR_CURRENTS_SPACESIPHON_MULT 4 //if the source of the air current is next to a turf open to space, the current will "consume" AIR_CURRENTS_SPACESIPHON_MULT times the air the src used in its balance. This allow for accounting of the fact most of the air you add in the turf connected to space, end up continuing into space
+var/global/datum/controller/process/aircurrents/air_currents_master
+/datum/controller/process/aircurrents/var/list/active_air_currents[0] //our list of active air currents
+/datum/controller/process/aircurrents/var/current_cycle = 0
+/datum/controller/process/aircurrents/var/sub_cycle = 0
```

Comment:
Any relative declaration complaints are for things like procs. Vars are allowed to be relatively pathed.